app/build.gradle

```diff
@@ -16,10 +17,13 @@ android {
         versionName "1.0"
 
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+
+        def url = properties.getProperty("url", "")
```

Comment:
API url - is not confidential data. No need to contain it in local file. Declare it here in build.gradle
And rename to WEATHER_API_URL